src/qibolab/instruments/qblox/controller.py

```diff
@@ -16,7 +16,20 @@
 from qibolab.sweeper import Parameter, Sweeper, SweeperType
 from qibolab.unrolling import Bounds
 
-SEQUENCER_MEMORY = 2**17
+MAX_NUM_BINS = 98304
+"""Maximum number of bins that should be used per sequencer in a readout
+module.
+
+One sequencer can have up to 2 ** 17 bins, however the 6 sequencers in
+module allocate bins from a shared memory, and this memory is smaller
+than 6 * 2 ** 17. Hence, if all sequencers are used at once, each cannot
+support 2 ** 17 bins. In fact, the total bin memory for the module is 3
+* (2 ** 17 + 2 ** 16). This means up to three sequencers used at once
```

Comment:
Is this number documented somewhere?

Reply:
nope, I learned it via private communication. But they will add explanation about this limitation in the documentation. Currently the documentation only mentions 2 ** 17 as being the max number of bins per each sequencer.